### ISSUE TITLE
`fetch` is catching errors occurring in component `render` methods

### ISSUE THREAD

**Issue by jimmed** [martyjs/marty#313]
Trying to describe this issue in a single sentence is incredibly hard.

Essentially, I have a pure (stateless) component being passed props from `Marty.createContainer` using its `fetch` property. If the fetch succeeds, but an error occurs later on in the Promise chain (i.e. rendering the inner component), this is caught by [`when`'s error handler][1].

This has two problems:
 - The error message is a little misleading. While the description is correct, at a first glance it suggests that the fetch has gone awry.
 - The errors are lumped together into a single object, with the key being the name of the failed fetch, and the value being the error. When viewed in Chrome's console it becomes hard/impossible to get to the stack trace without leaving breakpoints inside `when`.

As a proposed solution, how about iterating over these and logging each error individually? I'm not sure this is ideal, but it would make debugging things a little easier.

[1]: https://github.com/martyjs/marty/blob/master/lib/store/when.js#L33-L47

**Comment by jhollingworth** [comment]
If we removed that try/catch will that resolve you're issue? It's a hangover from darker days when we were trying to be clever with errors, I don't see much value any more

**Comment by jimmed** [comment]
That might be the best solution! Let me have a little play around quickly and see how that affects things for us.

**Comment by jimmed** [comment]
@jhollingworth removing the try/catch block is a slight improvement, as it means the error message is no longer misleading. Unfortunately, we still have the issue of being unable to get at the stack trace without a breakpoint/`debugger` statement.

Instead of removing the `catch` block entirely, how about making the error handling a little more verbose, like so?
```
try {
  //...
} catch(e) {
  // ...
  Object.keys(e).forEach(key => log.error(e[key]));
}
```

**Comment by goldensunliu** [comment]
https://github.com/martyjs/marty/issues/295 is also related

**Comment by sporto** [comment]
Yes please remove try/catches, we are finding this kind of issues all over, application errors ends up as something different

**Comment by jhollingworth** [comment]
OK, for now I'm going to just remove the try/catch. I will look into opportunities to improve this in the future

**Comment by jhollingworth** [comment]
Released in v0.9.16
